src/main/resources/META-INF/reaktivity/amqp.idl

```diff
@@ -17,9 +17,57 @@ scope amqp
 {
     scope control
     {
+        struct AmqpRouteEx
+        {
+            string linkTargetAddress;
+            uint8 linkRole;
+        }
     }
 
     scope stream
     {
+        struct AmqpBeginEx
+        {
+            string containerId;
+            uint16 channelNumber;
+            string targetAddress;
+            uint8 role;
+        }
+
+        struct AmqpDataEx
+        {
+            // AmqpTransferDataArguments
+            uint32 handle;
+            uint32 deliveryId;
+            octets[32] deliveryTag;
+            uint32 messageFormat;
+            uint8 settled;
+
+            // AmqpTransferDataMessageAnnotations
+            // TODO: need to add message annotations
+
+            // AmqpTransferDataMessageProperties
+            string messageId;
+            string to;
+            int64 creationTime;
```

Comment:
If properties is  general purpose, perhaps we need to represent that here as a list of name-value pairs?
Same for annotations?